src/screens/WalletConnect/WalletConnectCallRequest.js

```diff
@@ -95,10 +109,18 @@ class WalletConnectCallRequestScreen extends React.Component<Props, State> {
       if (matchingAssets && matchingAssets.length) {
         asset = matchingAssets[0]; // eslint-disable-line
         symbol = asset.symbol; // eslint-disable-line
-        amount = utils
-          .bigNumberify(data.substring(73))
-          .mul(utils.bigNumberify('10').pow(asset.decimals));
+        const iface = new Interface(ERC20_CONTRACT_ABI);
```

Comment:
Pls check how all this work with tokens those have a non-standard amount of decimals (not 18)

Reply:
That's a good catch. Do we have that information on the assets list?

Reply:
Actually, the contract doesn't seem to include the decimals and then we're doing `amount = parseFloat(utils.formatUnits(methodValue, asset.decimals));`. Is there any example you can come up with where we could have a problem regarding the decimal points?

Reply:
Everything should be fine here, I just missed that part during review: `amount = parseFloat(utils.formatUnits(methodValue, asset.decimals));`

Reply:
Changed this to string as last decimals are being rounded when converted to JavaScript number.